src/pip/__pip-runner__.py

```diff
@@ -18,9 +18,9 @@ def version_str(version):  # type: ignore
 
 if sys.version_info[:2] < PYTHON_REQUIRES:
     raise SystemExit(
-        "This version of pip does not support python {} (requires >={}).".format(
-            version_str(sys.version_info[:2]), version_str(PYTHON_REQUIRES)
-        )
+        "This version of pip does not support python "
+        f"{version_str(sys.version_info[:2])} "
+        f"(requires >={version_str(PYTHON_REQUIRES)})."
     )
```

Comment:
We can't use f-strings in this file, as the syntax must remain python 2 compatible.

Reply:
I've fixed and updated ruff config to never try to apply an upgrade rule to this file (only one upgrade rule it applied at the moment, but someone could manually run them or more could be added in the future)